sdks/java/harness/build.gradle

```diff
@@ -18,35 +18,44 @@
 
 plugins { id 'org.apache.beam.module' }
 
-// We specifically enumerate all the projects that we depend on since
-// the list is used in both defining the included set for the uber jar
-// and also the set of project level dependencies.
-def dependOnShadedProjects = [":model:pipeline", ":model:fn-execution", ":sdks:java:core"]
-def dependOnProjects = [":sdks:java:fn-execution",
-                        ":runners:core-java", ":runners:core-construction-java"]
+configurations {
+  provided
+}
+
+dependencies {
+  // provided dependencies are declared early so they can be resolved in shadowClosure
+  provided project(path: ":sdks:java:core", configuration: "shadow")
+}
 
 applyJavaNature(
   classesTriggerCheckerBugs: [
     'AssignWindowsRunner': 'https://github.com/typetools/checker-framework/issues/3794',
     'WindowMergingFnRunner': 'https://github.com/typetools/checker-framework/issues/3794',
   ],
   automaticModuleName: 'org.apache.beam.fn.harness',
-  validateShadowJar: false,
   testShadowJar: true,
+  shadowJarValidationExcludes: [
+    "io/github/classgraph/**",
+    "nonapi/io/github/classgraph/**",
+    "org/apache/beam/fn/**",
+    "org/apache/beam/model/**",
+    "org/apache/beam/runners/**",
+    "org/apache/beam/sdk/**",
+    "org/checkerframework/**",
+    "org/github/jamm/**",
```

Comment:
I think jamm should be provided as it needs to be on the boot agent classpath

Reply:
Done.